src/circular_buffer.rs

```diff
@@ -0,0 +1,313 @@
+
+#![deny(missing_docs)]
```

Comment:
nit: if you have these `#![deny]`s in `src/lib.rs`, the root of the crate, you don't need to write them here as the `#![deny]` in the root of the crate will apply to the whole crate.